intro-to-object-oriented-programming/defining-classes.md

```diff
@@ -0,0 +1,495 @@
+
+This variable is almost always named `self`. Technically, this variable can have any name, but most Pythonists will use `self`.
+
+`self` must **always** be the first parameter to any instance method. `__init__` is an instance method. We'll see more examples of this when exploring instance methods.
+
+### !callout-info
+
+## An Analogy for `self`: The Pronoun "I"
+
+In the English language, there is the pronoun "I." Each person uses the pronoun "I" to describe themselves. However, "I" doesn't point to the same person in every context; "I" always means the person who is saying it!
+
+Exactly like that, the variable `self` will exist in every instance. However, the value of `self` will always mean the instance itself.
+
+### !end-callout
+
+## Attributes: Variables on `self`
+
+Recall that **attributes** are pieces of data. In classes, they are pieces of data that become attached to each instance. Attributes maintain _state_. We want to write generic attributes in our class definitions.
+
+All attributes have names. The values of attributes will differ between instances.
+
+Here are some examples of attributes that could exist on classes:
+
+| Class  | Attribute                    | Instance Example                                                                      | Value of Attribute for Example |
+| ------ | ---------------------------- | ------------------------------------------------------------------------------------- | ------------------------------ |
+| Driver | Total amount of money earned | A driver with ID `DR0004`, 3 rides, who has earned $35, has an average rating of 4.67 | `35`                           |
+| Album  | Release date                 | Dirty Computer                                                                        | `"2018"`                       |
+
+Because attributes have a close relationship to an instance, attributes will look and act exactly like variables... but they are assigned through `self`. Our syntax for reading and writing attributes will include `self`.
+
+Read through this example. Then, read through the next sections to read about it in detail.
+
+```python
+class Album:
+
+    def __init__(self):
+        self.title = "Purple Rain"
+        print(f"We can also read the value of the title attribute")
+        print(f"by writing self.title: {self.title}")
+```
+
+When we make an instance of this class `Album`, it will print this:
+
+```
+We can also read the value of the title attribute
+by writing self.title: Purple Rain
+```
+
+### Writing Attributes Inside of a Class
+
+Inside of a class definition, we can assign values to an attribute with this syntax:
+
+```python
+self.attribute_name = new_value
+```
+
+... where `attribute_name` is the name of the attribute, and `new_value` is the new value.
```

Comment:
```suggestion
self.attribute_name = value_expression
```

... where `self` was the name we gave to the required first parameter of the instance method, `attribute_name` is the name of the attribute, and `new_value` is the new value.
```

@anselrognlie for visibility

Reply:
Oh